feat: Add support for SurfaceTintColor (#930)
unoplatform/Uno.Themes: doc/material-controls-extensions.md

# Material Control Extensions

## Icon

This feature allows for the addition of icon on the supported controls. Those icons could be any of the [`IconElement`](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.iconelement)s: `<BitmapIcon />`, `<FontIcon />`, `<PathIcon />` or `<SymbolIcon />`.

Here are supported control with samples:

* TextBox:

>   ``` xml
>     <TextBox Style="{StaticResource MaterialFilledTextBoxStyle}">
>         <um:ControlExtensions.Icon>
>             <SymbolIcon Symbol="SolidStar" />
>         </um:ControlExtensions.Icon>
>     </ComboBox>
>   ```

Reviewer: align content with the ```; you are currently padding the code with 2space
```suggestion
  ```xml
  <TextBox Style="{StaticResource MaterialFilledTextBoxStyle}">
      <um:ControlExtensions.Icon>
          <SymbolIcon Symbol="SolidStar" />
      </um:ControlExtensions.Icon>
  </ComboBox>
  ```
```
same for other places

            <PathIcon Data="{StaticResource Icon_more_vertical}" />
        </um:ControlExtensions.AlternateContent>
    </ToggleButton>
```

## Elevation

> This feature allows for setting the level of elevation to depict on the supported control.

Reviewer: ```suggestion
This feature allows to set the level of elevation to depict on the supported control.
```

> Setting the elevation on [supported controls]() can result in changes to both the shadow and the [surface tint](#surface-tint).

Reviewer: missing link

[Material Design Elevation Guidance](https://m3.material.io/styles/elevation/overview)

## Surface Tint

The Surface Tint properties allow for customization of how elevation can be depicted on certain controls. While the Background of a control remains static, the surface color can change based on the level of elevation.

### TintedBackground

This is a readonly property that will provide a SolidColorBrush depicting the control's current background color overlayed with the Surface Tint color at a certain opacity based on the elevation of the control.

### IsTintEnabled

> This feature allows for enabling or disabling the surface tint that may be applied to an elevated control. When `IsTintEnabled` is `false`, the `TintedBackground` property will remain the same value as the control's background color.

Reviewer: In `## Surface Tint`, you were using "Surface Tint" and "Background", here they are normal case. 
TitleCase or normal case, you got to pick a side.

### Example Usage for `Button`

The `ElevatedButtonStyle` in Uno Material supports elevation and surface tints through the use of the `TintedBackground`, `IsTintEnabled`, and `Elevation` attached properties.

`ElevatedButtonStyle` contains the following `Setter`s:

```xml
...

<Setter Property="um:ControlExtensions.Elevation"
        Value="1" />
<Setter Property="um:ControlExtensions.IsTintEnabled"
        Value="True" />
...
```

Within the `ControlTemplate` of the `ElevatedButtonStyle`, instead of performing a `TemplateBinding` to the `Background` property of the `Button`, we instead bind to the `TintedBackground` attached property:

> (Additional properties on `Grid` are omitted for brevity)
> 
> ```xml
> ...
> 
> <Grid x:Name="Root"
>       Background="{Binding Path=(um:ControlExtensions.TintedBackground), RelativeSource={RelativeSource TemplatedParent}}">
> 
>       ...
> 
> </Grid>
> 
> ...

Reviewer: ```suggestion
```xml
<Grid x:Name="Root"
      ...
      Background="{Binding Path=(um:ControlExtensions.TintedBackground), RelativeSource={RelativeSource TemplatedParent}}">
      <!-- content... -->
</Grid>
```
